cumulusci/cli/cci.py

```diff
@@ -807,15 +807,50 @@ def org_connect(config, org_name, sandbox, login_url, default, global_org):
         scope="web full refresh_token",
     )
     oauth_dict = oauth_capture()
+    # if sandbox:
+    #     org_config = ScratchOrgConfig(oauth_dict, org_name)
+    # else:
+    #     org_config = OrgConfig(oauth_dict, org_name)
+    #     org_config.load_userinfo()
+    #   unsure whether this logic is sound due to the fact that trialforce orgs are sandboxes
+    #   Need to validate this with someone. <-------------------***********NOTE***************
+    
     org_config = OrgConfig(oauth_dict, org_name)
     org_config.load_userinfo()
-
-    config.keychain.set_org(org_config, global_org)
-
+    if sandbox:
+        username = str(vars(org_config)["config"]["userinfo"]["preferred_username"])
+        result = sfdx(f"force:org:display -u {username}" )
+        config_value = (
+            io.TextIOWrapper(result.stdout, encoding=sys.stdout.encoding).read()
+        )
+        parsed = config_value.split("\n")
+        # skipping first two header rows
+        for i in range(3,len(parsed)-1):
+            value = parsed[i].split(' ')
+            if value[0].lower() == "expiration" and value[len(value)-1] != '':
+                org_config = ScratchOrgConfig(oauth_dict,org_name)
+                org_config.config.update({"username": username}) 
```

Comment:
This isn't a good way to get the org info because it will only work for orgs that sfdx already knows about, and the whole reason for this feature is to be able to connect an org that someone else created and neither sfdx nor cci know about yet.

Instead, run `org_config._load_orginfo()` and then look at the values in org_config.organization_sobject which includes the values documented in https://developer.salesforce.com/docs/atlas.en-us.object_reference.meta/object_reference/sforce_api_objects_organization.htm

Reply:
Make sure you add "scratch": True to the config dict when creating the ScratchOrgConfig instance. This is needed so that it knows to use ScratchOrgConfig instead of OrgConfig when the org is loaded back from the keychain in the future.

Reply:
Yeah I was going to change this parsing not proud of this work. Why I didn't want to raise a pr yet. Will implement the change.

Reply:
@jofsky If it's a scratch org I think we'll need to add it to the sfdx keychain too, in addition to the cci keychain. I haven't used it before but the auth:sfdxurl:store sfdx command (https://developer.salesforce.com/docs/atlas.en-us.sfdx_cli_reference.meta/sfdx_cli_reference/cli_reference_force_auth.htm#cli_reference_force_auth) looks like it could help with that. You have to create a temporary file containing `force://<refreshToken>@<instanceUrl>` (using the refresh_token and instance_url from the org_config) and then call the force:auth:sfdxurl command referencing that file. See how this goes and we can talk about it...